admin/test/test_installer.py

```diff
@@ -502,13 +458,23 @@ def upload_docker_compose_files():
             steps=repeat(1, 60)
         )
 
-        # This isn't in the tutorial, but docker-compose doesn't retry failed
-        # pulls and pulls fail all the time.
+        def cleanup_container(ignored):
+            self.addCleanup(
+                remote_docker_compose,
+                self.client_node_ip,
+                self.docker_host,
+                self.compose_node1.path,
+                'down'
+            )
+        d.addCallback(cleanup_container)
+
+        # docker-compose doesn't retry failed pulls and pulls fail all the
+        # time.
         def pull_postgres():
-            return remote_docker(
+            return remote_docker_compose(
                 self.client_node_ip,
                 self.docker_host,
-                'pull', 'postgres:latest'
+                self.compose_node1.path, 'pull'
```

Comment:
Using `docker-compose pull` ensures that we're pulling the same version of the image that's used in the config file and pulls to all the docker swarm hosts.